metricflow/model/validations/unique_valid_name.py

```diff
@@ -60,6 +82,14 @@ def check_valid_name(  # noqa: D
                     f"({TimeGranularity.list_names()})",
                 )
             )
+        if name.upper() in {reserved_name.value for reserved_name in MetricFlowReservedKeywords}:
```

Comment:
We've been going with lower case as per `LowerCaseNamesRule`

Reply:
Will change. We use upper case in the SQL reserved keywords checks, so I followed that because this is a reserved keyword, but now that I think of it we always reference it lower case ourselves since it's an element name.